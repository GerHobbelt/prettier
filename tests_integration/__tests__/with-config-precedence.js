"use strict";

const runPrettier = require("../runPrettier");

test("CLI overrides take precedence without --config-precedence", () => {
  const output = runPrettier("cli/config/", ["--print-width", "1", "**/*.js"]);
  expect(output.stdout).toMatchSnapshot();
  expect(output.status).toEqual(0);
});

test("CLI overrides take precedence with --config-precedence cli-override", () => {
  const output = runPrettier("cli/config/", [
    "--print-width",
    "1",
    "--config-precedence",
    "cli-override",
    "**/*.js"
  ]);
  expect(output.stdout).toMatchSnapshot();
  expect(output.status).toEqual(0);
});

test("CLI overrides take lower precedence with --config-precedence file-override", () => {
  const output = runPrettier("cli/config/js/", [
    "--tab-width",
    "1",
    "--config-precedence",
    "file-override",
    "**/*.js"
  ]);
  expect(output.stdout).toMatchSnapshot();
  expect(output.status).toEqual(0);
});

test("CLI overrides are still applied when no config is found with --config-precedence file-override", () => {
  const output = runPrettier("cli/config/no-config/", [
    "--tab-width",
    "6",
    "--config-precedence",
    "file-override",
    "**/*.js"
  ]);
  expect(output.stdout).toMatchSnapshot();
  expect(output.status).toEqual(0);
});

test("CLI overrides gets ignored when config exists with --config-precedence prefer-file", () => {
  const output = runPrettier("cli/config/js/", [
    "--print-width",
    "1",
    "--tab-width",
    "1",
    "--config-precedence",
    "prefer-file",
    "**/*.js"
  ]);
  expect(output.stdout).toMatchSnapshot();
  expect(output.status).toEqual(0);
});

test("CLI overrides gets applied when no config exists with --config-precedence prefer-file", () => {
  const output = runPrettier("cli/config/no-config/", [
    "--print-width",
    "1",
    "--tab-width",
    "7",
    "--no-config",
    "--config-precedence",
    "prefer-file",
    "**/*.js"
  ]);
  expect(output.stdout).toMatchSnapshot();
  expect(output.status).toEqual(0);
});
