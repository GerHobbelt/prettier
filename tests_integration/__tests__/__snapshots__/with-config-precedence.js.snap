// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI overrides are still applied when no config is found with --config-precedence file-override 1`] = `
"function noConfigJs() {
      console.log(\\"no-config/file.js should have no semicolons\\")
}
"
`;

exports[`CLI overrides gets applied when no config exists with --config-precedence prefer-file 1`] = `
"function noConfigJs() {
       console.log(
              \\"no-config/file.js should have no semicolons\\"
       );
}
"
`;

exports[`CLI overrides gets ignored when config exists with --config-precedence prefer-file 1`] = `
"function js() {
        console.log(\\"js/file.js should have tab width 8\\");
}
\\"use strict\\";

module.exports = {
        tabWidth: 8
};
"
`;

exports[`CLI overrides take lower precedence with --config-precedence file-override 1`] = `
"function js() {
        console.log(\\"js/file.js should have tab width 8\\");
}
\\"use strict\\";

module.exports = {
        tabWidth: 8
};
"
`;

exports[`CLI overrides take precedence with --config-precedence cli-override 1`] = `
"console.log(
  \\"jest/__best-tests__/file.js should have semi\\"
);
console.log(
  \\"jest/Component.js should not have semi\\"
)
console.log(
  \\"jest/Component.test.js should have semi\\"
);
function js() {
        console.log(
                \\"js/file.js should have tab width 8\\"
        );
}
\\"use strict\\";

module.exports = {
        tabWidth: 8
};
function noConfigJs() {
  console.log(
    \\"no-config/file.js should have no semicolons\\"
  )
}
function packageJs() {
   console.log(
      \\"package/file.js should have tab width 3\\"
   );
}
function rcJson() {
  console.log.apply(
    null,
    [
      'rc-json/file.js',
      'should have trailing comma',
      'and single quotes',
    ],
  );
}
function rcYaml() {
  console.log.apply(
    null,
    [
      'rc-yaml/file.js',
      'should have trailing comma',
      'and single quotes',
    ],
  );
}
"
`;

exports[`CLI overrides take precedence without --config-precedence 1`] = `
"console.log(
  \\"jest/__best-tests__/file.js should have semi\\"
);
console.log(
  \\"jest/Component.js should not have semi\\"
)
console.log(
  \\"jest/Component.test.js should have semi\\"
);
function js() {
        console.log(
                \\"js/file.js should have tab width 8\\"
        );
}
\\"use strict\\";

module.exports = {
        tabWidth: 8
};
function noConfigJs() {
  console.log(
    \\"no-config/file.js should have no semicolons\\"
  )
}
function packageJs() {
   console.log(
      \\"package/file.js should have tab width 3\\"
   );
}
function rcJson() {
  console.log.apply(
    null,
    [
      'rc-json/file.js',
      'should have trailing comma',
      'and single quotes',
    ],
  );
}
function rcYaml() {
  console.log.apply(
    null,
    [
      'rc-yaml/file.js',
      'should have trailing comma',
      'and single quotes',
    ],
  );
}
"
`;
